dev-ops/migrations/create_gene_sets_for_outlier_analyses.js

```diff
@@ -0,0 +1,185 @@
+// to run:
+// mongo MedBook < create_gene_sets_for_outlier_analyses.js
+//
+// Mission:
+// Create three gene sets for each outlier anaysis so that:
+// 1. we can call GSEA from the outlier analysis and pass it the _id of
+//    a gene set object
+// 2. we can switch to the Handsontable viewer for outlier analysis results
+//    in the future
+//
+// Also add a couple summary output variables to each job so we don't have
+// to load the whole (large) output object to the client when listing
+// the jobs that they have access to:
+// - up_genes_count
+// - down_genes_count
+// - top5percent_genes_count
+
+print("Removing old migration data");
+
+var geneSetsQuery = {
+  "associated_object.collection_name": "Jobs"
+};
+var recordIds = db.gene_sets.distinct("_id", geneSetsQuery);
+db.records.remove({
+  "associated_object.collection_name": "GeneSets",
+  "associated_object.mongo_id": { $in: recordIds },
+});
+db.gene_sets.remove(geneSetsQuery);
+
+
+function createGeneSet(job, outlier_type, fields, prependToName, description) {
+  var dataArray = job.output[outlier_type + "_genes"];
+
+  // calculate the gene_labels from the data array
+  var gene_labels = [];
+  for (var i = 0; i < dataArray.length; i++) {
+    gene_labels.push(dataArray[i].gene_label);
+  }
+
+  // generate a random ID for the new gene set
+  var geneSetId = (new ObjectId()).valueOf();
+
+  var name = prependToName + ": " + job.args.sample_label;
+  if (outlier_type !== "top5percent") {
+    name += " vs. " + job.args.sample_group_name;
+  }
+
+  db.gene_sets.insert({
+    _id: geneSetId,
+    name: name,
+    description: description,
+
+    associated_object: {
+      collection_name: "Jobs",
+      mongo_id: job._id
+    },
+    metadata: {
+      outlier_type: outlier_type
+    },
+
+    fields: fields,
+    gene_labels: gene_labels,
+    gene_label_field: "Gene",
+  });
+
+  var bulk = db.records.initializeUnorderedBulkOp();
+  for (i = 0; i < dataArray.length; i++) {
+    var oldRecord = dataArray[i];
+
+    var record = {
+      "Gene": oldRecord.gene_label,
+      "Sample value": oldRecord.sample_value,
+      "Background median": oldRecord.background_median,
```

Comment:
Should this be omitted for top5% records? right now they all have `"Background median" : undefined,`